File: Examples/FxSocket.Ftp/Windows/CS/FtpDesktopClientDemo/ConnectForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using FxSocket.Net;

namespace FxSocketSamples
{
    public partial class ConnectForm : Form
    {
        public ConnectForm()
        {
            InitializeComponent();


            Server = (string)Common.GetKey("Server", "");
            Port = Common.GetKey("Port", 21);
            SslMode = (SslMode)Common.GetKey("Ssl", 0);

            UserName = (string)Common.GetKey("User", "");
            Password = (string)Common.GetKey("Pass", "");
            UseUtf8Encoding = Common.GetKey("Utf8", false);
            PassiveMode = Common.GetKey("Passive", true);

            Ccc = Common.GetKey("Ccc", false);
            ClientCert = (string)Common.GetKey("Cert", "");

            Protocol = (TlsVersion)Common.GetKey("Protocol", (int)TlsVersion.Any);
            AllowedSuite = (TlsCipherSuite)Common.GetKey("AllowedSuite", (long)TlsCipherSuite.All);

            ProxyHost = (string)Common.GetKey("ProxyHost", "");
            ProxyPort = Common.GetKey("ProxyPort", 1080);
            ProxyType = (FtpProxyType)Common.GetKey("ProxyType", 0);
            ProxyAuthnMethod = (FtpProxyAuthentication)Common.GetKey("ProxyAuthnMethod", 0);
            ProxyUser = (string)Common.GetKey("ProxyUser", "");
            ProxyPassword = (string)Common.GetKey("ProxyPassword", "");
            ProxyDomain = (string)Common.GetKey("ProxyDomain", "");
        }

        public string Server
        {
            get { return tbServer.Text; }
            set { tbServer.Text = value; }
        }

        public int Port
        {
            get { return (int)tbPort.Value; }
            set { tbPort.Value = value; }
        }

        public SslMode SslMode
        {
            get { return (SslMode)cbxSslMode.SelectedIndex; }
            set { cbxSslMode.SelectedIndex = (int)value; }
        }

        public string UserName
        {
            get { return tbUserName.Text; }
            set { tbUserName.Text = value; }
        }

        public string Password
        {
            get { return tbPassword.Text; }
            set { tbPassword.Text = value; }
        }

        public bool UseUtf8Encoding
        {
            get { return cbUtf8Encoding.Checked; }
            set { cbUtf8Encoding.Checked = value; }
        }

        public bool PassiveMode
        {
            get { return cbPasv.Checked; }
            set { cbPasv.Checked = value; }
        }

        #region Security

        public bool Ccc
        {
            get { return cbClearCommandChannel.Checked; }
            set { cbClearCommandChannel.Checked = value; }
        }

        public string ClientCert
        {
            get { return tbCert.Text; }
            set { tbCert.Text = value; }
        }

        public TlsCipherSuite AllowedSuite
        {
            get
            {
                if (suiteDl.SelectedIndex == 0)
                    return TlsCipherSuite.All;
                else
                    return TlsCipherSuite.Secure;
            }
            set
            {
                if (value == TlsCipherSuite.All)
                    suiteDl.SelectedIndex = 0;
                else
                    suiteDl.SelectedIndex = 1;
            }
        }

        public TlsVersion Protocol
        {
            get
            {
                TlsVersion value = 0;
                if (cbTls10.Checked)
                    value |= TlsVersion.TLS10;
                if (cbTls11.Checked)
                    value |= TlsVersion.TLS11;
                if (cbTls12.Checked)
                    value |= TlsVersion.TLS12;

                if (cbSsl30.Checked)
                    value |= TlsVersion.SSL30;

                return value;
            }
            set
            {
                cbTls10.Checked = ((value & TlsVersion.TLS10) != 0);
                cbTls11.Checked = ((value & TlsVersion.TLS11) != 0);
                cbTls12.Checked = ((value & TlsVersion.TLS12) != 0);

                cbSsl30.Checked = ((value & TlsVersion.SSL30) != 0);
            }
        }

        #endregion

        #region Proxy

        public string ProxyHost
        {
            get { return tbProxyHost.Text; }
            set { tbProxyHost.Text = value; }
        }

        public int ProxyPort
        {
            get { return (int)tbProxyPort.Value; }
            set { tbProxyPort.Value = value; }
        }

        public string ProxyUser
        {
            get { return tbProxyUser.Text; }
            set { tbProxyUser.Text = value; }
        }

        public string ProxyPassword
        {
            get { return tbProxyPassword.Text; }
            set { tbProxyPassword.Text = value; }
        }

        public string ProxyDomain
        {
            get { return tbProxyDomain.Text; }
            set { tbProxyDomain.Text = value; }
        }

        public FtpProxyType ProxyType
        {
            get { return (FtpProxyType)cbxProxyType.SelectedIndex; }
            set { cbxProxyType.SelectedIndex = (int)value; }
        }

        public FtpProxyAuthentication ProxyAuthnMethod
        {
            get { return (FtpProxyAuthentication)cbxProxyMethod.SelectedIndex; }
            set { cbxProxyMethod.SelectedIndex = (int)value; }
        }

        #endregion

        private void connectBtn_Click(object sender, EventArgs e)
        {
            Common.SetKey("Server", Server);
            Common.SetKey("Port", Port);
            Common.SetKey("Ssl", (int)SslMode);

            Common.SetKey("User", UserName);
            Common.SetKey("Pass", Password);
            Common.SetKey("Utf8", UseUtf8Encoding);
            Common.SetKey("Passive", PassiveMode);

            Common.SetKey("Ccc", Ccc);
            Common.SetKey("Cert", ClientCert);

            Common.SetKey("Protocol", (int)Protocol);
            Common.SetKey("AllowedSuite", (int)AllowedSuite);

            Common.SetKey("ProxyHost", ProxyHost);
            Common.SetKey("ProxyPort", ProxyPort);
            Common.SetKey("ProxyType", ProxyType);
            Common.SetKey("ProxyAuthnMethod", ProxyAuthnMethod);
            Common.SetKey("ProxyUser", ProxyUser);
            Common.SetKey("ProxyPassword", ProxyPassword);
            Common.SetKey("ProxyDomain", ProxyDomain);
        }

        private void sslDropDown_SelectedIndexChanged(object sender, EventArgs e)
        {
            btCertBrowse.Enabled = tbCert.Enabled = cbSsl30.Enabled = cbTls10.Enabled = cbTls11.Enabled = cbTls12.Enabled = suiteDl.Enabled = cbxSslMode.SelectedIndex > 0;
        }

        private void certBrowseBtn_Click(object sender, EventArgs e)
        {
            try
            {
                OpenFileDialog dlg = new OpenFileDialog();
                dlg.Title = "Select a private key file";
                dlg.FileName = tbCert.Text;
                dlg.Filter = "All files|*.*";
                dlg.FilterIndex = 1;
                if (dlg.ShowDialog() == DialogResult.OK)
                {
                    tbCert.Text = dlg.FileName;
                }
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message, "Error");
            }
        }
    }
}

File: Examples/FxSocket.Ftp/Windows/VB/FtpDesktopClientDemo/ConnectForm.vb
Imports System.ComponentModel
Imports System.Text
Imports FxSocket.Net

Namespace FxSocketSamples
	Partial Public Class ConnectForm
		Inherits Form
		Public Sub New()
			InitializeComponent()


			Server = CStr(Common.GetKey("Server", ""))
			Port = Common.GetKey("Port", 21)
			SslMode = CType(Common.GetKey("Ssl", 0), SslMode)

			UserName = CStr(Common.GetKey("User", ""))
			Password = CStr(Common.GetKey("Pass", ""))
			UseUtf8Encoding = Common.GetKey("Utf8", False)
			PassiveMode = Common.GetKey("Passive", True)

			Ccc = Common.GetKey("Ccc", False)
			ClientCert = CStr(Common.GetKey("Cert", ""))

			Protocol = CType(Common.GetKey("Protocol", CInt(TlsVersion.Any)), TlsVersion)
			AllowedSuite = CType(Common.GetKey("AllowedSuite", CLng(TlsCipherSuite.All)), TlsCipherSuite)

			ProxyHost = CStr(Common.GetKey("ProxyHost", ""))
			ProxyPort = Common.GetKey("ProxyPort", 1080)
			ProxyType = CType(Common.GetKey("ProxyType", 0), FtpProxyType)
			ProxyAuthnMethod = CType(Common.GetKey("ProxyAuthnMethod", 0), FtpProxyAuthentication)
			ProxyUser = CStr(Common.GetKey("ProxyUser", ""))
			ProxyPassword = CStr(Common.GetKey("ProxyPassword", ""))
			ProxyDomain = CStr(Common.GetKey("ProxyDomain", ""))
		End Sub

		Public Property Server() As String
			Get
				Return tbServer.Text
			End Get
			Set(ByVal value As String)
				tbServer.Text = value
			End Set
		End Property

		Public Property Port() As Integer
			Get
				Return CInt(Fix(tbPort.Value))
			End Get
			Set(ByVal value As Integer)
				tbPort.Value = value
			End Set
		End Property

		Public Property SslMode() As SslMode
			Get
				Return CType(cbxSslMode.SelectedIndex, SslMode)
			End Get
			Set(ByVal value As SslMode)
				cbxSslMode.SelectedIndex = CInt(value)
			End Set
		End Property

		Public Property UserName() As String
			Get
				Return tbUserName.Text
			End Get
			Set(ByVal value As String)
				tbUserName.Text = value
			End Set
		End Property

		Public Property Password() As String
			Get
				Return tbPassword.Text
			End Get
			Set(ByVal value As String)
				tbPassword.Text = value
			End Set
		End Property

		Public Property UseUtf8Encoding() As Boolean
			Get
				Return cbUtf8Encoding.Checked
			End Get
			Set(ByVal value As Boolean)
				cbUtf8Encoding.Checked = value
			End Set
		End Property

		Public Property PassiveMode() As Boolean
			Get
				Return cbPasv.Checked
			End Get
			Set(ByVal value As Boolean)
				cbPasv.Checked = value
			End Set
		End Property

		#Region "Security"

		Public Property Ccc() As Boolean
			Get
				Return cbClearCommandChannel.Checked
			End Get
			Set(ByVal value As Boolean)
				cbClearCommandChannel.Checked = value
			End Set
		End Property

		Public Property ClientCert() As String
			Get
				Return tbCert.Text
			End Get
			Set(ByVal value As String)
				tbCert.Text = value
			End Set
		End Property

		Public Property AllowedSuite() As TlsCipherSuite
			Get
				If suiteDl.SelectedIndex = 0 Then
					Return TlsCipherSuite.All
				Else
					Return TlsCipherSuite.Secure
				End If
			End Get
			Set(ByVal value As TlsCipherSuite)
				If value = TlsCipherSuite.All Then
					suiteDl.SelectedIndex = 0
				Else
					suiteDl.SelectedIndex = 1
				End If
			End Set
		End Property

		Public Property Protocol() As TlsVersion
			Get
				Dim value As TlsVersion = 0
				If cbTls10.Checked Then
					value = value Or TlsVersion.TLS10
				End If
				If cbTls11.Checked Then
					value = value Or TlsVersion.TLS11
				End If
				If cbTls12.Checked Then
					value = value Or TlsVersion.TLS12
				End If

				If cbSsl30.Checked Then
					value = value Or TlsVersion.SSL30
				End If

				Return value
			End Get
			Set(ByVal value As TlsVersion)
				cbTls10.Checked = ((value And TlsVersion.TLS10) <> 0)
				cbTls11.Checked = ((value And TlsVersion.TLS11) <> 0)
				cbTls12.Checked = ((value And TlsVersion.TLS12) <> 0)

				cbSsl30.Checked = ((value And TlsVersion.SSL30) <> 0)
			End Set
		End Property

		#End Region

		#Region "Proxy"

		Public Property ProxyHost() As String
			Get
				Return tbProxyHost.Text
			End Get
			Set(ByVal value As String)
				tbProxyHost.Text = value
			End Set
		End Property

		Public Property ProxyPort() As Integer
			Get
				Return CInt(Fix(tbProxyPort.Value))
			End Get
			Set(ByVal value As Integer)
				tbProxyPort.Value = value
			End Set
		End Property

		Public Property ProxyUser() As String
			Get
				Return tbProxyUser.Text
			End Get
			Set(ByVal value As String)
				tbProxyUser.Text = value
			End Set
		End Property

		Public Property ProxyPassword() As String
			Get
				Return tbProxyPassword.Text
			End Get
			Set(ByVal value As String)
				tbProxyPassword.Text = value
			End Set
		End Property

		Public Property ProxyDomain() As String
			Get
				Return tbProxyDomain.Text
			End Get
			Set(ByVal value As String)
				tbProxyDomain.Text = value
			End Set
		End Property

		Public Property ProxyType() As FtpProxyType
			Get
				Return CType(cbxProxyType.SelectedIndex, FtpProxyType)
			End Get
			Set(ByVal value As FtpProxyType)
				cbxProxyType.SelectedIndex = CInt(value)
			End Set
		End Property

		Public Property ProxyAuthnMethod() As FtpProxyAuthentication
			Get
				Return CType(cbxProxyMethod.SelectedIndex, FtpProxyAuthentication)
			End Get
			Set(ByVal value As FtpProxyAuthentication)
				cbxProxyMethod.SelectedIndex = CInt(value)
			End Set
		End Property

		#End Region

		Private Sub connectBtn_Click(ByVal sender As Object, ByVal e As EventArgs) Handles connectBtn.Click
			Common.SetKey("Server", Server)
			Common.SetKey("Port", Port)
			Common.SetKey("Ssl", CInt(SslMode))

			Common.SetKey("User", UserName)
			Common.SetKey("Pass", Password)
			Common.SetKey("Utf8", UseUtf8Encoding)
			Common.SetKey("Passive", PassiveMode)

			Common.SetKey("Ccc", Ccc)
			Common.SetKey("Cert", ClientCert)

			Common.SetKey("Protocol", CInt(Protocol))
			Common.SetKey("AllowedSuite", CInt(AllowedSuite))

			Common.SetKey("ProxyHost", ProxyHost)
			Common.SetKey("ProxyPort", ProxyPort)
			Common.SetKey("ProxyType", ProxyType)
			Common.SetKey("ProxyAuthnMethod", ProxyAuthnMethod)
			Common.SetKey("ProxyUser", ProxyUser)
			Common.SetKey("ProxyPassword", ProxyPassword)
			Common.SetKey("ProxyDomain", ProxyDomain)
		End Sub

		Private Sub sslDropDown_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles cbxSslMode.SelectedIndexChanged
			suiteDl.Enabled = cbxSslMode.SelectedIndex > 0
			cbTls12.Enabled = suiteDl.Enabled
			cbTls11.Enabled = cbTls12.Enabled
			cbTls10.Enabled = cbTls11.Enabled
			cbSsl30.Enabled = cbTls10.Enabled
			tbCert.Enabled = cbSsl30.Enabled
			btCertBrowse.Enabled = tbCert.Enabled
		End Sub

		Private Sub certBrowseBtn_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btCertBrowse.Click
			Try
				Dim dlg As New OpenFileDialog()
				dlg.Title = "Select a private key file"
				dlg.FileName = tbCert.Text
				dlg.Filter = "All files|*.*"
				dlg.FilterIndex = 1
				If dlg.ShowDialog() = System.Windows.Forms.DialogResult.OK Then
					tbCert.Text = dlg.FileName
				End If
			Catch ex As Exception
				MessageBox.Show(ex.Message, "Error")
			End Try
		End Sub
	End Class
End Namespace
